src/Components/Web.Extensions/src/InputFile/FileListEntryStream.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal abstract class FileListEntryStream : Stream
+    {
+        private long _position;
+
+        protected FileListEntry File { get; }
+
+        protected FileListEntryStream(FileListEntry file)
+        {
+            File = file;
+        }
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => false;
+
+        public override long Length => File.Size;
+
+        public override long Position
+        {
+            get => _position;
+            set => throw new NotSupportedException();
+        }
+
+        public override void Flush()
+            => throw new NotSupportedException();
+
+        public override int Read(byte[] buffer, int offset, int count)
+            => throw new NotSupportedException("Synchronous reads are not supported.");
+
+        public override long Seek(long offset, SeekOrigin origin)
+            => throw new NotSupportedException();
+
+        public override void SetLength(long value)
+            => throw new NotSupportedException();
+
+        public override void Write(byte[] buffer, int offset, int count)
+            => throw new NotSupportedException();
+
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            var maxBytesToRead = (int)Math.Min(count, Length - Position);
+
+            if (maxBytesToRead == 0)
```

Comment:
Can we have this say ``` if (Length == Position) { return 0;}```? Returning a 0 from a Stream indicates EOF, so let's be explicit about when we think we're done. Can we also add the usual bound checks for `Read` : https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L335-L342

Reply:
How about something like this:

```csharp
int maxBytesToRead = (int)(Length - Position);

if (maxBytesToRead > count)
{
    maxBytesToRead = count;
}

if (maxBytesToRead <= 0)
{
    return 0;
}
```

Since it closely matches https://github.com/dotnet/runtime/blob/ead2cd50799ecc523e23f50e50b54d8a30d0aa4f/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L346-L350?